cub/util_ptx.cuh

```diff
@@ -192,6 +200,45 @@ __device__ __forceinline__ void BFI(
 }
 
 
+/**
+ * \brief Bitfield insert for 64-bit types.
+ */
+template <typename UnsignedBits>
+__device__ __forceinline__ void BFI(
+    UnsignedBits            &ret,
+    UnsignedBits            x,
+    UnsignedBits            y,
+    unsigned int            bit_start,
+    unsigned int            num_bits,
+    Int2Type<8>      /*byte_len*/)
+{
+#if CUB_PTX_ARCH >= 200
+    asm ("bfi.b64 %0, %1, %2, %3, %4;" :
+        "=l"((unsigned long long int) ret) : "l"((unsigned long long int) y), "l"((unsigned long long int) x), "r"(bit_start), "r"(num_bits));
+#else
+    x <<= bit_start;
+    unsigned long long int MASK_X = ((1 << num_bits) - 1) << bit_start;
+    unsigned long long int MASK_Y = ~MASK_X;
```

Comment:
Please use `std::uint64_t`, etc, instead of variable-width types like `unsigned long long int`.